pkg/result/processors/severity.go

```diff
@@ -64,29 +63,30 @@ func (p *Severity) Process(issues []result.Issue) ([]result.Issue, error) {
 		return issues, nil
 	}
 
-	return transformIssues(issues, func(issue *result.Issue) *result.Issue {
-		if issue.Severity != "" && !p.override {
-			return issue
-		}
-
-		for _, rule := range p.rules {
-			rule := rule
+	return transformIssues(issues, p.transform), nil
+}
 
-			ruleSeverity := p.defaultSeverity
-			if rule.severity != "" {
-				ruleSeverity = rule.severity
+func (p *Severity) transform(issue *result.Issue) *result.Issue {
+	for _, rule := range p.rules {
+		if rule.match(issue, p.files, p.log) {
+			if rule.severity == severityFromLinter || rule.severity == "" && p.defaultSeverity == severityFromLinter {
```

Comment:
please, add parentheses (`( )`)

Reply:
I meant

```go
if rule.severity == severityFromLinter || (rule.severity == "" && p.defaultSeverity == severityFromLinter) {
```

expression in simple but really long
I wasted some time to process it :D

Reply:
https://github.com/golangci/golangci-lint/pull/4492